tests/ghost_test.go

```diff
@@ -0,0 +1,178 @@
+package tests
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/bytebase/bytebase/resources/mysql"
+	"github.com/github/gh-ost/go/base"
+	"github.com/github/gh-ost/go/logic"
+	ghostsql "github.com/github/gh-ost/go/sql"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+type config struct {
+	host           string
+	port           int
+	user           string
+	password       string
+	database       string
+	table          string
+	alterStatement string
+	execute        bool
+}
+
+func newMigrationContext(config config) (*base.MigrationContext, error) {
+	migrationContext := base.NewMigrationContext()
+	migrationContext.InspectorConnectionConfig.Key.Hostname = config.host
+	migrationContext.InspectorConnectionConfig.Key.Port = config.port
+	migrationContext.CliUser = config.user
+	migrationContext.CliPassword = config.password
+	migrationContext.DatabaseName = config.database
+	migrationContext.OriginalTableName = config.table
+	migrationContext.AlterStatement = config.alterStatement
+	// set defaults
+	migrationContext.AllowedRunningOnMaster = true
+	migrationContext.ConcurrentCountTableRows = true
+	migrationContext.HooksStatusIntervalSec = 60
+	migrationContext.ReplicaServerId = 99999
+	migrationContext.CutOverType = base.CutOverAtomic
+
+	if migrationContext.AlterStatement == "" {
+		return nil, fmt.Errorf("alterStatement must be provided and must not be empty")
+	}
+	parser := ghostsql.NewParserFromAlterStatement(migrationContext.AlterStatement)
+	migrationContext.AlterStatementOptions = parser.GetAlterStatementOptions()
+
+	if migrationContext.DatabaseName == "" {
+		if parser.HasExplicitSchema() {
+			migrationContext.DatabaseName = parser.GetExplicitSchema()
+		} else {
+			return nil, fmt.Errorf("database must be provided and database name must not be empty, or alterStatement must specify database name")
+		}
+	}
+	if migrationContext.OriginalTableName == "" {
+		if parser.HasExplicitTable() {
+			migrationContext.OriginalTableName = parser.GetExplicitTable()
+		} else {
+			return nil, fmt.Errorf("table must be provided and table name must not be empty, or alterStatement must specify table name")
+		}
+	}
+	migrationContext.Noop = !config.execute
+	migrationContext.ServeSocketFile = fmt.Sprintf("/tmp/gh-ost.%s.%s.sock", migrationContext.DatabaseName, migrationContext.OriginalTableName)
```

Comment:
my 2 cent is to put logically related setting together, e.g. the code related to DatabaseName should be grouped together. Otherwise, it's just scattered around.

Another option is to put simple set together, however, for this one seems to be a bit hard.

Reply:
Moved `Noop` up.